definitions/npm/ramda_v0.21.x/flow_v0.28.x-v0.30.x/ramda_v0.21.x.js

```diff
@@ -500,8 +500,8 @@ declare module ramda {
   declare function dissocPath<T>(key: Array<string>, val:T, ...args: Array<void>): (src: {[k:string]:T}) => {[k:string]:T};
   declare function dissocPath<T>(key: Array<string>, val:T, src: {[k:string]:T}): {[k:string]:T};
 
-  declare function evolve<V,R>(fn: NestedObject<(x:any) => R>, ...rest: Array<void>): (src: NestedObject<any>) => NestedObject<R>;
-  declare function evolve<V,R>(fn: NestedObject<(x:any) => R>, src: NestedObject<any>): NestedObject<R>;
+  declare function evolve<R: NestedObject<any>>(fn: NestedObject<UnarySameTypeFn<any>>, ...rest: Array<void>): (src: R) => R;
```

Comment:
That's absolutely not true.

I've used this definition in my own code.  This definition will pass a typecheck even if the input and result are not referentially transparent.

There is one test already that demonstrates exactly this use-case, and which you can see passed CI.

https://github.com/ajhyndman/flow-typed/blob/ramda-evolve-definition/definitions/npm/ramda_v0.21.x/flow_v0.28.x-v0.30.x/test_ramda_v0.21.x_object.js#L30